### PR TITLE
[Product Collection] Fix: HTML entity decoding for product names in Hand-Picked Products

### PR DESCRIPTION
**Pull request by imanish003**
## What

I've implemented three updates in this PR:

- **Resolved the HTML Entities Decoding Issue:**
    - For instance, where it should say “Hoodie – Black”, it incorrectly displayed `Hoodie &#8211; Black`. The screenshot below illustrates this issue.

image

- **Label Change for Consistency:**
    - Updated the label from “Pick some products” to “Hand-picked Products” for alignment with the filters dropdown.
- **Simplified by Removing the Description:**
    - I omitted the description as it wasn't significantly beneficial.

Fixes #

## Why

These changes were proposed by @manospsyx  on Slack (p1700652774848299-slack-C05UMTA7W0P).

## Testing Instructions

_Please consider any edge cases this change may have, and also other areas of the product this may impact._

**Checking the Hyphen Representation**:

1. Add a hyphen `–` to the name of any product, like **`Hoodie - Black`**.
2. Insert a Product Collection block into a new post.
3. Apply the “Hand-picked Products” filter in Inspector Controls.
4. Search for the product with the hyphen and confirm it displays correctly, as shown below:

image

5. Feel free to test other [HTML entities](https://developer.mozilla.org/en-US/docs/Glossary/Entity) too.

**Verifying Label Update**:

- Ensure the text reads “HAND-PICKED PRODUCTS” instead of “PICK SOME PRODUCTS”, as depicted in the image below:

image

* [ ] Do not include in the Testing Notes 
* [ ] Should be tested by the development team exclusively 

## Screenshots or screencast

| Before | After |
| ------ | ----- |
|        |       |

## WooCommerce Visibility

Required:

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental
* [ ] N/A

## Checklist

Required:

* [x] This PR has either a `[type]` label or a `[skip-changelog]` label.
* [ ] This PR is assigned to a milestone.

Conditional:

* [x] This PR has a UI change and has been cross-browser tested at different viewport sizes on both the frontend and in the editor.
* [ ] This PR has a changelog description (if `[skip-changelog]` label is not present).
* [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR adds/removes an experimental interfaces, and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR has been accessibility tested.
* [ ] This PR has had any necessary documentation added/updated.

## Changelog

> [Product Collection] Fix: HTML entity decoding for product names in Hand-Picked Products

### CODE REVIEW

File: assets/js/blocks/product-collection/inspector-controls/hand-picked-products-control.tsx [hunk 1]
```diff
@@ -17,6 +17,25 @@ import {
...
+/**
+ * Decodes HTML entities in a string.
+ * Example:
+ * decodeHTMLEntities( 'foo &amp; bar' ) // 'foo & bar'
+ * decodeHTMLEntities( 'Hoodie &#8211; Black' ) // 'Hoodie – Black'
+ *
+ * @param {string} str - The string containing HTML entities.
+ * @return {string} - The decoded string.
+ */
+function decodeHTMLEntities( str?: string ) {
```
Comment (albarin) [inline assets/js/blocks/product-collection/inspector-controls/hand-picked-products-control.tsx]: Hey @imanish003 I think we already have a function that does this, check [here](https://github.com/woocommerce/woocommerce-blocks/blob/4020db4c10d4a3d984e2207285976434bb0b2a89/assets/js/atomic/blocks/product-elements/add-to-cart/product-types/variable/variation-attributes/attribute-select-control.tsx#L78). I think we should probably used that.

Comment (imanish003) [reply assets/js/blocks/product-collection/inspector-controls/hand-picked-products-control.tsx]: Good catch @albarin. Thanks 🙏  Let me quickly change the code.

Comment (imanish003) [reply assets/js/blocks/product-collection/inspector-controls/hand-picked-products-control.tsx]: Made the changes in https://github.com/woocommerce/woocommerce-blocks/pull/11927/commits/70ea5f1a679dd94a685cef39631e406574130d44

**Review by albarin — APPROVED** [review]
Thanks for the update! LGTM 🚀

**Review by kmanijak — APPROVED** [review]
Thanks for a quick fix 🚀 I tested some other entities as well, works as expected!

**Pull request merged**